coding-guides/starboard.md

```diff
@@ -0,0 +1,281 @@
+      const imageLink = attachment.split('.');
+      const typeOfImage = imageLink[imageLink.length - 1];
+      const image = /(jpg|jpeg|png|gif)/gi.exec(typeOfImage);
+      if (!image) return '';
+      return attachment;
+    };
+  }
+}
+```
+
+Alright, let's break this down. 
+
+```js
+this.starEmbed = async (color, description, author, authorURL, timestamp, footer, image) => {
+  const embed = { 
+    'color': color, 
+    'description': description, 
+    'author': { 
+    'name': author,
+      'icon_url': authorURL
+    },
+    'image': { 
+      'url': image 
+    }, 
+    'timestamp': timestamp, 
+    'footer': { 
+      'text': footer 
+    } 
+  };
+  return embed; 
+};
+
+this.extension = async (reaction, attachment) => {
+  const imageLink = attachment.split('.');
+  const typeOfImage = imageLink[imageLink.length - 1];
+  const image = /(jpg|jpeg|png|gif)/gi.exec(typeOfImage);
+  if (!image) return '';
+  return attachment;
+};
+```
+
+This creates 2 functions attached to `this.` for later convenience. `this.starEmbed` will create the embed for us to send. Next, we create a function that will determine whether or not there is an image attached to the message.
+
+Let's keep going.
+
+```js
+async run(reaction, user) {
+  const message = reaction.message;
+  const { starboardChannel } = this.client.settings.get(message.guild.id)
+  if (reaction.emoji.name !== '⭐') return;
+  if (message.author.id === user.id) return message.channel.send(`${user}, you cannot star your own messages.`);
+  if (message.author.bot) return message.channel.send(`${user}, you cannot star bot messages.`);
+}
+```
+
+This is some simple setup for later on. For ease, I define message as `message`, but this is completely optional. Next, we grab the starboardChannel key from the guilds settings. Then, we preform a couple of checks on the reaction and the message. First, we check if the reaction is the unicode star emote. Then, we preform two checks on the message, which are checking if the user who added the star is the author of the message, and then we check if the author of the message is a bot. If neither of these checks return true, then we're good to go. Let's move on.
+
+```js
+async run(reaction, user) {
+  const message = reaction.message;
+  const { starboardChannel } = this.client.settings.get(message.guild.id)
+  if (reaction.emoji.name !== '⭐') return;
+  if (message.author.id === user.id) return message.channel.send(`${user}, you cannot star your own messages.`);
+  if (message.author.bot) return message.channel.send(`${user}, you cannot star bot messages.`);
+  try {
+    const fetch = await message.guild.channels.find('name', starboardChannel).fetchMessages({ limit: 100 });
+    const stars = fetch.find(m => m.embeds[0].footer.text.startsWith('⭐') && m.embeds[0].footer.text.endsWith(message.id));
+    if (stars) {
+      const star = /\⭐\s([0-9]{1,3})\s\|\s([0-9]{17,20})/g.exec(stars.embeds[0].footer.text);
```

Comment:
You need a single match, remove the global flag. Also, you don't need to escape the emoji. And put `^` at the start of it so it doesn't try to check the string anywhere.